From owner-linux-pci@atrey.karlin.mff.cuni.cz Tue Apr 18 02:21:16 2006
From: Ingo Oeser <netdev@axxeo.de>
To: Greg Kroah-Hartmann <gregkh@suse.de>
Subject: PCI: Documentation: no more device ids
Date: Tue, 18 Apr 2006 11:20:55 +0200
Cc: Jeff Garzik <jgarzik@pobox.com>
Content-Disposition: inline
Message-Id: <200604181120.55783.netdev@axxeo.de>

From: Ingo Oeser <netdev@axxeo.de>

Document that we don't like to add more PCI device ids
but are happy to accept PCI vendor ids for linux/include/pci_ids.h

Original text from Jeff Garzik.

Signed-off-by: Ingo Oeser <netdev@axxeo.de>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 Documentation/pci.txt |   12 +++++++++++-
 1 file changed, 11 insertions(+), 1 deletion(-)

--- gregkh-2.6.orig/Documentation/pci.txt
+++ gregkh-2.6/Documentation/pci.txt
@@ -259,7 +259,17 @@ on the bus need to be capable of doing i
 to be handled by platform and generic code, not individual drivers.
 
 
-8. Obsolete functions
+8. Vendor and device identifications
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+For the future, let's avoid adding device ids to include/linux/pci_ids.h.
+
+PCI_VENDOR_ID_xxx for vendors, and a hex constant for device ids.
+
+Rationale:  PCI_VENDOR_ID_xxx constants are re-used, but device ids are not.
+    Further, device ids are arbitrary hex numbers, normally used only in a
+    single location, the pci_device_id table.
+
+9. Obsolete functions
 ~~~~~~~~~~~~~~~~~~~~~
 There are several functions which you might come across when trying to
 port an old driver to the new PCI interface.  They are no longer present
